src/shared/bus-unit-util.c

```diff
@@ -148,6 +148,26 @@ int bus_append_unit_property_assignment(sd_bus_message *m, const char *assignmen
 
                 r = sd_bus_message_append(m, "sv", field, "t", bytes);
                 goto finish;
+        } else if (streq(field, "TasksMax")) {
+                uint64_t t;
+
+                if (isempty(eq) || streq(eq, "infinity"))
+                        t = (uint64_t) -1;
```

Comment:
setting the actual property to (uint64_t) -1 means disabling any configuraiton of it. If TasksMax is initialized like that it basically tells us: "don't bother with the pids cgroup controller at all". We do not clamp this into the actually valid range simply because it's precisely supposed to be outside of it.

We do this the same way for all other cgroup properties, for example MemoryLimit= is also stored as -1 internally if it is disabled.